ctapipe/reco/sklearn.py

```diff
@@ -965,7 +967,14 @@ def __call__(self, telescope_type, table):
                     cv_values.mean(),
                     cv_values.std(),
                 )
-        self.cv_predictions[telescope_type] = vstack(predictions)
+
+        if self.output_path:
+            write_table(
+                vstack(predictions),
+                self.output_path,
+                f"/cv_predictions_{telescope_type}",
+                overwrite=True,
```

Comment:
And here you set it to true immediately

Reply:
You can move the writing into the loop (further reducing memory) by adding `append=True` here after overwriting the file in `__init__`